db/migrations/25-03-07-compute-contribution-stats.sql

```diff
@@ -0,0 +1,133 @@
+                    'UNAPPROVED' as state
+                from phrase
+                where deleted_by is not null and deleted_at is not null
+            )
+        ) log
+        join phrase_word phw on phw.phrase_id = log.phrase_id
+        join phrase ph on ph.id = log.phrase_id
+        where log.updated_at > date_bin('7 days', date_trunc('day', $1), timestamp '2024-12-15')
+            and log.updated_at < date_bin('7 days', date_trunc('day', $1 + interval '7 days'), timestamp '2024-12-15')
+        order by ph.language_id, phw.word_id, log.updated_at desc
+        ) changed_phrase
+    left join lateral (
+        select h.* from (
+            (
+                select phrase_id, updated_by, updated_at, gloss, state
+                from gloss
+                where exists (
+                        select 1 from phrase_word phw
+                        join phrase ph on ph.id = phw.phrase_id
+                        where ph.id = gloss.phrase_id
+                            and ph.language_id = changed_phrase.language_id
+                            and phw.word_id = changed_phrase.word_id
+                    )
+                    and gloss.updated_at < date_bin('7 days', date_trunc('day', $1 - interval '7 days'), timestamp '2024-12-15')
+            ) union all (
+                select
+                    id as phrase_id,
+                    deleted_by as updated_by,
+                    deleted_at as updated_at,
+                    '' as gloss,
+                    'UNAPPROVED' as state
+                from phrase
+                where deleted_by is not null and deleted_at is not null
+                    and phrase.language_id = changed_phrase.language_id
+                    and exists (
+                        select 1 from phrase_word phw
+                        where phw.phrase_id = phrase.id
+                            and phw.word_id = changed_phrase.word_id
+                    )
+                    and phrase.deleted_at < date_bin('7 days', date_trunc('day', $1 - interval '7 days'), timestamp '2024-12-15')
+            )
+        ) h
+        order by h.updated_at desc
+        limit 1
+    ) prev_phrase on true
+    where changed_phrase.updated_by is not null
+    group by changed_phrase.language_id, changed_phrase.updated_by
+    on conflict (language_id, user_id, week)
+    do update set
+        approved_count = excluded.approved_count,
+        revoked_count = excluded.revoked_count,
+        edited_approved_count = excluded.edited_approved_count,
+        edited_unapproved_count = excluded.edited_unapproved_count;
+$$;
+
+select cron.schedule(
+    'Generate Weekly Contribution Statistics',
+    '5 0 * * 0',
+    'select generate_weekly_contribution_stats(now());'
+);
+
+select generate_weekly_contribution_stats(week) from (
+    select timestamp '2024-02-04' + interval '7 days' * generate_series(0, floor(extract(day from now() - timestamp '2024-02-04') / 7) - 1) AS week
+) series;
```

Comment:
This sets up a weekly cron job to compute these stats, so that they can be easily fetched on demand with low response time